scripts/generate-docs.js

```diff
@@ -0,0 +1,128 @@
+  const hooksWithSchemas = hooks.filter((hook) => HookSchemas[hook])
+  return $t`
+    ${
+      tasksWithSchemas.length
+        ? formatSchemas(
+            'Tasks',
+            tasksWithSchemas.map((task) => ({
+              name: task,
+              schema: TaskSchemas[task].describe((TaskSchemas[task].description ?? '') + '\n### Task options')
+            }))
+          )
+        : ''
+    }
+    ${
+      hooksWithSchemas.length
+        ? formatSchemas(
+            'Hooks',
+            hooksWithSchemas.map((hook) => ({
+              name: hook,
+              schema: HookSchemas[hook].describe((HookSchemas[hook].description ?? '') + '\n### Hook options')
+            }))
+          )
+        : ''
+    }
+    ${
+      PluginSchemas[pluginPackage]
+        ? formatSchemas('Plugin-wide options', [
+            {
+              name: pluginPackage,
+              schema: PluginSchemas[pluginPackage]
+            }
+          ])
+        : ''
+    }
+	`
+}
+
+const EMPTY_OBJECT_SCHEMA = `#### Task options
+
+| Property | Type |
+| :------- | :--- |
+
+_All properties are optional._`
+
+function postProcessMarkdown(markdown) {
+  let replaced = markdown
+    .replace(/^#/gm, '##')
+    .replace(/_Object containing the following properties:_\n\n/g, '')
+    .replaceAll(EMPTY_OBJECT_SCHEMA, '')
+
+  // for more complex types, the autogenerated schema docs aren't that readable.
+  // we can hide them by adding an unclosed HTML commment in the description.
+  if (replaced.includes('<!--') && !replaced.includes('-->')) {
+    replaced += '-->\n'
+  }
+
+  return replaced
+}
+
+const BEGIN_COMMENT = '<!-- begin autogenerated docs -->\n'
+const END_COMMENT = '<!-- end autogenerated docs -->'
+
+function replaceBetween(text, replaceWith, startMarker, endMarker) {
+  const startIndex = text.indexOf(startMarker)
+  if (startIndex === -1) {
+    throw new Error(`Start marker "${startMarker}" not found`)
+  }
+
+  const endIndex = text.indexOf(endMarker, startIndex + startMarker.length)
+  if (endIndex === -1) {
+    throw new Error(`End marker "${endMarker}" not found after start marker "${startMarker}"`)
+  }
+
+  return text.slice(0, startIndex + startMarker.length) + replaceWith + text.slice(endIndex)
+}
+
+async function main() {
+  const plugins = await fs.readdir('plugins')
+
+  return await Promise.all(
+    plugins.map(async (plugin) => {
+      const readmePath = path.join('plugins', plugin, 'readme.md')
+      const generatedOptionsMarkdown = await formatPluginSchemas(plugin)
+
+      const originalReadme = await fs.readFile(readmePath, 'utf-8')
+
+      try {
+        const replacedReadme = replaceBetween(
+          originalReadme,
+          generatedOptionsMarkdown,
+          BEGIN_COMMENT,
+          END_COMMENT
+        )
+
+        await fs.writeFile(readmePath, replacedReadme, 'utf-8')
+        console.log(`written ${readmePath}`)
+      } catch (e) {
+        console.error(`no replacement markers in ${readmePath}`)
```

Comment:
**question**: do we want to prevent a push by exiting with an error code here?

Reply:
"no replacement markers" is fine, that just means it's a plugin readme that doesn't have any autogenerated docs